src/components/views/settings/JoinRuleSettings.tsx

```diff
@@ -297,7 +321,22 @@ const JoinRuleSettings: React.FC<JoinRuleSettingsProps> = ({
                     {preferredKnockVersion && upgradeRequiredPill}
                 </>
             ),
-            description: _t("People cannot join unless access is granted."),
+            description: (
+                <>
+                    {_t("People cannot join unless access is granted.")}
+                    <LabelledCheckbox
+                        className="mx_JoinRuleSettings_labelledCheckbox"
+                        disabled={!!preferredKnockVersion || joinRule !== JoinRule.Knock}
```

Comment:
could you explain why this needs to be gated by `!!preferredKnockVersion` ?

Reply:
It is mainly because I want to avoid any interaction with the join rule when need to upgrade first.

![2023-09-07-11-55-35](https://github.com/matrix-org/matrix-react-sdk/assets/1422657/f3832ac3-a839-456d-b48a-d3db0b2babd1)

Reply:
Thanks for the catch! 🙂